db/models/comments.model.js

```diff
@@ -20,14 +20,21 @@ function insertComment(article_id, { author, body }) {
             return result.rows[0].body
         })
         .catch((err) => {
-            if (err.detail === 'Key (article_id)=(90) is not present in table "articles".') {
-                return Promise.reject({ status: 404, msg: 'article does not exist' })
-            }
-            else if (err.detail === 'Key (author)=(i_dont_exist) is not present in table "users".') {
-                return Promise.reject({ status: 404, msg: 'user does not exist' })
+            if (err.code === '23503') {
+                return Promise.reject({ status: 404, msg: 'Not found' })
```

Comment:
This is a mixure of error handling and ultimately be handled in an error handler. There is no reason to Promise.reject - if there is an error code it is already a rejected promise (from PSQL) so you should be using next in the catch and then sending the status code and message from the error handling middleware based on the code.